### PR TITLE
feat: Use feature config to enable/disable guest links SQSERVICES-1049

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/SQSERVICES-1049" title="SQSERVICES-1049" target="_blank">StorySQSERVICES-1049</a>  [iOS] Use feature config to enable/disable guest links
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

This is a new feature configuration that allows team admins to enabled / disable the guest-link feature for their team. Clients should rely on this configuration and hide / show the ability to create a guest link.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Source/Model/FeatureConfig/Feature.swift [hunk 1]
```diff
@@ -35,6 +35,7 @@ public class Feature: ZMManagedObject {
         case conferenceCalling
         case fileSharing
         case selfDeletingMessages
+        case guestLinks
```
Comment (johnxnguyen) [inline Source/Model/FeatureConfig/Feature.swift:38]: What do you think about calling it `conversationGuestLinks` to be more descriptive?

**Review by agisilaos — APPROVED** [review]

File: Source/Model/FeatureConfig/Feature.swift (hunk 1 above)
Comment (KaterinaWire) [reply Source/Model/FeatureConfig/Feature.swift]: I agree with this suggestion. 
Replaced.

**Review by johnxnguyen — APPROVED** [review]

**Comment by KaterinaWire** [comment]
release major

**Comment by zenkins** [comment]
Hello, @KaterinaWire! Merging PR for command **release major** for PR #**1295**. Will create a **major** release after merged.
Next version is 282.0.0.

**Pull request merged**